modular_skyrat/modules/engineering_override/code/engineering_override.dm

```diff
@@ -0,0 +1,137 @@
+ * Sensitive areas like the vault, command quarters, heads' offices, etc. are not applicable.
+ */
+/area/station/ai_monitored/command/storage/eva
+	engineering_override_eligible = TRUE
+
+/area/station/cargo
+	engineering_override_eligible = TRUE
+
+/area/station/command/bridge
+	engineering_override_eligible = TRUE
+
+/area/station/command/teleporter
+	engineering_override_eligible = TRUE
+
+/area/station/command/gateway
+	engineering_override_eligible = TRUE
+
+/area/station/construction/storage_wing
+	engineering_override_eligible = TRUE
+
+/area/station/engineering/atmos
+	engineering_override_eligible = TRUE
+
+/area/station/engineering/atmospherics_engine
+	engineering_override_eligible = TRUE
+
+/area/station/engineering/storage/tcomms
+	engineering_override_eligible = TRUE
+
+/area/station/hallway/secondary/service
+	engineering_override_eligible = TRUE
+
+/area/station/maintenance
+	engineering_override_eligible = TRUE
+
+/area/station/medical
+	engineering_override_eligible = TRUE
+
+/area/station/service
+	engineering_override_eligible = TRUE
+
+/area/station/science
+	engineering_override_eligible = TRUE
+
+/obj/machinery/door
+	/// Determines if engineers get access to this door on orange alert
+	var/engineering_override = FALSE
+
+/// Check for the three states of open access. Emergency, Unrestricted, and Engineering Override
+/obj/machinery/door/allowed(mob/Mob)
+	if(emergency)
+		return TRUE
+
+	if(unrestricted_side(Mob))
+		return TRUE
+
+	if(engineering_override)
+		var/mob/living/carbon/human/user = Mob
+		var/obj/item/card/id/card = user.get_idcard(TRUE)
+		if(istype(user))
+			if(ACCESS_ENGINEERING in card.access)
+				return TRUE
+	return ..()
+
+/// Activate the airlock overrides, called by the change of alert level
+/proc/enable_engineering_access()
+	for(var/area/station_area in get_areas(/area/station))
+		if(!station_area.engineering_override_eligible)
+			continue
+
+		for(var/obj/machinery/door/airlock/airlock in station_area)
+			airlock.engineering_override = TRUE
+			airlock.normalspeed = FALSE
+			airlock.update_icon(ALL, 0)
+
+	message_admins("Engineering override has been turned ON for station airlocks.")
+	minor_announce("Engineering staff will have expanded access to areas of the station during the emergency.", "Engineering Emergency")
+
+/// Disable the airlock overrides, called by the change of the alert level
+/proc/revoke_engineering_access()
+	for(var/area/station_area in get_areas(/area/station))
+		if(!station_area.engineering_override_eligible)
+			continue
+
+		for(var/obj/machinery/door/airlock/airlock in station_area)
+			airlock.engineering_override = FALSE
+			airlock.normalspeed = TRUE
+			airlock.update_icon(ALL, 0)
+	message_admins("Engineering override has been turned OFF for station airlocks.")
+
+/// Someone or the AI or silicons tries to change the access on the airlock
+/obj/machinery/door/airlock/proc/toggle_engineering(mob/user)
```

Comment:
I'm not sure what that documentation means, will people be able to actually toggle engineering access on airlocks themselves, without needing to be on orange alert to do so?

Reply:
Yes, we would add it to the AI/borg panel for interacting with airlocks.

Reply:
Is it added yet, in that case?

Reply:
Nope, so we'll remove it.